servers/account-data-deleter/src/dataService/importers/ImportBase.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  SendMessageBatchCommand,
+  SendMessageBatchRequestEntry,
+  SQSClient,
+} from '@aws-sdk/client-sqs';
+import { S3Bucket } from '../s3Service';
+import AdmZip from 'adm-zip';
+import path from 'path';
+import { sqs } from '../../aws/sqs';
+import { config } from '../../config';
+import { ImportMapping, ImportMessage } from './types';
+import { chunk } from 'lodash';
+import { nanoid } from 'nanoid';
+import { serverLogger } from '@pocket-tools/ts-logger';
+import * as Sentry from '@sentry/node';
+
+export abstract class ImportBase {
+  public sqsClient: SQSClient;
+  constructor(
+    public s3: S3Bucket,
+    sqsClient?: SQSClient,
+  ) {
+    if (sqsClient != null) {
+      this.sqsClient = sqsClient;
+    } else {
+      this.sqsClient = sqs;
+    }
+  }
+
+  abstract start(): Promise<void>;
+
+  /**
+   * Load an archive from an S3 key and extract
+   * the entries into memory
+   * @param fileKey the key of the s3 object
+   * @param prefix optional prefix for filtering which
+   * files in the archive are returned
+   * @returns
+   */
+  async loadArchive(
+    fileKey: string,
+    prefix?: string,
+  ): Promise<AdmZip.IZipEntry[]> {
+    const body = await this.s3.getObjectBody(fileKey);
+    if (body == null) {
+      return [];
+    } else {
+      const buffer = Buffer.from(await body.transformToByteArray());
+      const zip = new AdmZip(buffer);
+      const entries = zip.getEntries();
+      if (prefix) {
+        return entries.filter((entry) => {
+          return path.basename(entry.entryName).startsWith(prefix);
+        });
+      } else {
+        return entries;
+      }
+    }
+  }
+
+  /**
+   * Send messages to the import queue. Sends a batch of 10 messages, and
+   * each message payload has a batch of records of config.listImport.chunkSize
+   * length.
+   * @param records array of records which are, at minimum, urls to be imported
+   * @param formatter callback for formatting records into the expected message
+   * @param metadata additional metadata for logging and error reporting
+   */
+  async sendMessages<T extends keyof ImportMapping>(
+    records: ImportMapping[T],
+    formatter: (records: ImportMapping[T]) => ImportMessage<T>,
+    metadata: { fileKey: string; userId: string; type: T },
+  ) {
+    const batches = chunk(chunk(records, config.listImport.chunkSize), 10);
```

Comment:
Can we add a comment that this always can't be more then 10 after our SQS discovery